cmd/hamctl/command/rollback_test.go

```diff
@@ -0,0 +1,181 @@
+			}
+
+			err := json.NewEncoder(w).Encode(resp)
+			require.NoError(t, err)
+
+		default:
+			require.Fail(t, "http url was not found")
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+
+	c := internalhttp.Client{
+		BaseURL: server.URL,
+	}
+
+	gitConfigAPI := git.GitConfigAPIMock{
+		CommitterDetailsFunc: func() (string, string, error) {
+			return "name", "email", nil
+		},
+	}
+
+	releaseClient := actions.NewReleaseHttpClient(&gitConfigAPI, &c)
+
+	runCommand := func(selectRollback command.SelectRollbackRelease, t *testing.T, args ...string) ([]string, error) {
+		var output []string
+		cmd := command.NewRollback(&c, &serviceName, func(f string, args ...interface{}) {
+			output = append(output, fmt.Sprintf(f, args))
+		}, selectRollback, releaseClient)
+		cmd.SetArgs(args)
+		err := cmd.Execute()
+
+		return output, err
+	}
+
+	t.Run("test rollback to previous commit with no artifacts specified", func(t *testing.T) {
+		describeReleaseResponse = provideArtifacts(3)
+
+		var rollback command.SelectRollbackRelease = func(environment string, releases []internalhttp.DescribeReleaseResponseRelease) (int, error) {
+			return 1, nil
+		}
+
+		output, err := runCommand(rollback, t, "--env", "dev")
+
+		require.NoError(t, err)
+		assert.Equal(
+			t,
+			[]string{"[✓] Starting rollback of service [some-service-name artifact-1] to %!s(MISSING)\n", "[✓] [released]\n"},
+			output,
+		)
+	})
+
+	t.Run("test rollback to previous commit with no artifacts specified select index 0", func(t *testing.T) {
+		describeReleaseResponse = provideArtifacts(3)
+		var rollback command.SelectRollbackRelease = func(environment string, releases []internalhttp.DescribeReleaseResponseRelease) (int, error) {
+			return 0, nil
+		}
+
+		output, err := runCommand(rollback, t, "--env", "dev")
+
+		require.NoError(t, err)
+		assert.Equal(
+			t,
+			[]string{"[✓] Starting rollback of service [some-service-name artifact-0] to %!s(MISSING)\n", "[✓] [released]\n"},
+			output,
+		)
+	})
+
+	t.Run("test rollback to previous commit with no artifacts not enough releases", func(t *testing.T) {
+		describeReleaseResponse = provideArtifacts(1)
+		var rollback command.SelectRollbackRelease = func(environment string, releases []internalhttp.DescribeReleaseResponseRelease) (int, error) {
+			return 0, nil
+		}
+
+		_, err := runCommand(rollback, t, "--env", "dev")
+
+		require.ErrorContains(t, err, "can't do rollback")
+	})
+
+	t.Run("test rollback to previous commit with artifact specified", func(t *testing.T) {
+		describeReleaseResponse = provideArtifacts(3)
+		var rollback command.SelectRollbackRelease = func(environment string, releases []internalhttp.DescribeReleaseResponseRelease) (int, error) {
+			return -1, nil
+		}
+
+		output, err := runCommand(rollback, t, "--env", "dev", "--artifact", "artifact-0")
+
+		require.NoError(t, err)
+		assert.Equal(
+			t,
+			[]string{"[✓] Starting rollback of service [some-service-name artifact-0] to %!s(MISSING)\n", "[✓] [released]\n"},
```

Comment:
Do we really expect `%!s(MISSING)` to be written to the user? 🙈

Reply:
Missed ... on args in my formatter. Should've suspected that it was it...